### PR TITLE
Put block processing in a function and handle its errors with catch

### PR DESCRIPTION
**Pull request by F3Joule**

### CODE REVIEW

File: src/substrate/subscribe.ts [hunk 1]
```diff
@@ -22,6 +22,8 @@ async function main () {
   const waitNextBlock = async () =>
     new Promise(resolve => setTimeout(resolve, blockTime))
 
+  const continueOnFail : boolean = JSON.parse(process.env.SUBSTRATE_CONTINUE_ON_FAIL) || true
```
Comment (siman) [inline src/substrate/subscribe.ts:25]:
```suggestion
  const continueOnFail: boolean = JSON.parse(process.env.SUBSTRATE_CONTINUE_ON_FAIL) || true
```

File: src/substrate/subscribe.ts [hunk 2]
```diff
@@ -72,7 +74,7 @@ async function main () {
       log.debug('Waiting for the best finalized block...')
       await waitNextBlock()
       bestFinalizedBlock = await getBestFinalizedBlock()
-      continue
+      return;
```
Comment (siman) [inline src/substrate/subscribe.ts:77]:
```suggestion
      return
```

File: src/substrate/subscribe.ts [hunk 3]
```diff
@@ -123,6 +125,19 @@ async function main () {
...
+  while (true) {
+    try {
+      await processBlock()
+    } catch {
+      if (!continueOnFail) {
+        log.error('Failed to process block')
+        break;
+      } else {
+        continue;
+      }
```
Comment (siman) [inline src/substrate/subscribe.ts:138]:
```suggestion
      }
```

File: src/substrate/subscribe.ts [hunk 4]
```diff
@@ -22,6 +22,11 @@ async function main () {
   const waitNextBlock = async () =>
     new Promise(resolve => setTimeout(resolve, blockTime))
 
+  const dotenvContinueOnFail =  process.env.SUBSTRATE_CONTINUE_ON_FAIL
+  const continueOnFail = typeof dotenvContinueOnFail === 'undefined'
+    ? true
+    : dotenvContinueOnFail !== 'false'
```
Comment (subsocialdev) [inline src/substrate/subscribe.ts:28]:
```suggestion
  function parseBool (x: boolean | string): boolean {
    if (typeof x === 'boolean') return x

    return nonEmptyStr(x) ? ['true', 'yes'].indexOf(x.trim().toLowerCase()) >= 0 : false
  }

  const continueOnFail = parseBool(process.env.SUBSTRATE_CONTINUE_ON_FAIL)
```

File: localhost.env [hunk 5]
```diff
@@ -4,6 +4,8 @@ SUBSTRATE_URL=ws://127.0.0.1:9944
 # Substrate events config
 SUBSTRATE_EVENT_SECTIONS=social
 SUBSTRATE_EVENT_METHODS=BlogCreated,BlogUpdated,BlogFollowed,BlogUnfollowed,AccountFollowed,AccountUnfollowed,PostCreated,PostUpdated,PostShared,CommentCreated,CommentUpdated,PostReactionCreated,CommentReactionCreated,ProfileCreated
+# Must be a boolean
```
Comment (subsocialdev) [inline localhost.env:7]:
```suggestion

# Must be a boolean
```

**Comment by siman** [comment]
It's better to ignore failing substrate events via .env file

**Pull request closed without merging**